src/datasets/utils/_hf_hub_fixes.py

```diff
@@ -60,6 +66,112 @@ def create_repo(
         )
 
 
+def get_repo_id_from_repo_url(repo_url: Union[str, Any]) -> str:
+    """
+    In 0.12.0 the output of `huggingface_hub.hf_api.create_repo` change output from `str` containing the
+    repo_url, to `RepoUrl` object. This function checks the huggingface_hub version to get the repo_id from
+    the repo_url, in the correct way.
+
+    Args:
+        repo_url (`str` or `huggingface_hub.hf_api.RepoUrl`): URL to the repo.
+
+    Returns:
+        `str`: repo_id, the ID of the repository to push to in the following format: `<user>/<dataset_name>` or
+            `<org>/<dataset_name>`.
+    """
+    if version.parse(huggingface_hub.__version__) < version.parse("0.12.0"):
+        from urllib.parse import urlparse
+
+        repo_id = urlparse(repo_url).path[1:]
```

Comment:
This line is still wrong for repo_type = "dataset" (it is only valid for repo_type=None or "model"):
- for repo_url = "https://hub-ci.huggingface.co/datasets/__DUMMY_TRANSFORMERS_USER__/test-16766285471927"
- it gives repo_id = 'datasets/__DUMMY_TRANSFORMERS_USER__/test-16766285471927'
- instead of: '__DUMMY_TRANSFORMERS_USER__/test-16766285471927'